reporting/api/src/main/java/org/jboss/windup/reporting/rules/CreateApplicationListReportRuleProvider.java

```diff
@@ -69,15 +77,51 @@ private void createIndexReport(GraphContext context)
         report.setReportFilename(OUTPUT_FILENAME);
 
         GraphService<WindupVertexListModel> listService = new GraphService<>(context, WindupVertexListModel.class);
-        WindupVertexListModel<ApplicationReportModel> applications = listService.create();
-        for (ApplicationReportModel applicationReportModel : applicationReportService.findAll())
+        WindupVertexListModel<ApplicationReportModel> appsListVertex = listService.create();
+
+
+        final Iterable<ApplicationReportModel> apps = applicationReportService.findAll();
+        List<ApplicationReportModel> appsList = new ArrayList();
+        for (ApplicationReportModel applicationReportModel : apps)
         {
             if (applicationReportModel.isMainApplicationReport() != null && applicationReportModel.isMainApplicationReport())
-                applications.addItem(applicationReportModel);
+                appsList.add(applicationReportModel);
         }
+        // Our current model doesn't keep the list order, but once I wrote, I'm leaving the sorting here for when it does.
+        appsList.sort(new AppRootFileNameComparator());
+
+        for (ApplicationReportModel applicationReportModel : appsList)
+            appsListVertex.addItem(applicationReportModel);
+
+        // Related data map
         Map<String, WindupVertexFrame> relatedData = new HashMap<>();
-        relatedData.put("applications", applications);
+        relatedData.put("applications", appsListVertex);
 
         report.setRelatedResource(relatedData);
     }
+
+
+
+    private static class AppRootFileNameComparator implements Comparator<ApplicationReportModel>
+    {
+        public int compare(ApplicationReportModel o1, ApplicationReportModel o2)
+        {
+            // If the info is missing, put that to the end. This may be the case of virtual apps.
+            if (null == o1.getProjectModel() || null == o1.getProjectModel().getRootFileModel() || null == o1.getProjectModel().getRootFileModel().getFileName() )
+                return 1;
+            if (null == o2.getProjectModel() || null == o2.getProjectModel().getRootFileModel() || null == o2.getProjectModel().getRootFileModel().getFileName() )
+                return -1;
+
+            // On error resume next.
+            try {
+                return ObjectUtils.compare(
+                    o1.getProjectModel().getRootFileModel().getFileName(),
+                    o2.getProjectModel().getRootFileModel().getFileName());
+            }
+            catch (Throwable ex)
```

Comment:
I changed this to be haltOnException = true, so I wanted to be super-defensive. If the sorting fails for whichever reason, it's better to show unsorted app list than no app list and fail :)